canonical-web-and-design/maas-ui
# feat(ui): build PowerIcon component

## Done

- Built `PowerIcon` component and used in machine list, machine details header and LXD VM list
- Converted `PowerColumn` to TypeScript

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the machine list and check that the power icons show properly
- Open the power action dropdown and change the power state of a machine.
- Check that a spinner shows while the power changes.
- Go to a machine details page and check the same things in the header.
- Go to the LXD VM list and check that the power icons show correctly.

## Fixes

Fixes #2410
